dolphinscheduler-task-plugin/dolphinscheduler-task-java/src/main/java/org/apache/dolphinscheduler/plugin/task/java/JavaTask.java

```diff
@@ -0,0 +1,422 @@
+                .append(mainJarName).append(" ")
+                .append(javaParameters.getMainArgs().trim()).append(" ")
+                .append(javaParameters.getJvmArgs().trim());
+        return builder.toString();
+    }
+
+    private String getResourceNameOfMainJar(ResourceInfo mainJar) {
+        if (null == mainJar) {
+            throw new RuntimeException("The jar for the task is required.");
+        }
+        return mainJar.getId() == 0
+                ? mainJar.getRes()
+                // when update resource maybe has error
+                : mainJar.getResourceName().replaceFirst(SINGLE_SLASH, "");
+    }
+
+    @Override
+    public void cancelApplication(boolean cancelApplication) throws Exception {
+        // cancel process
+        shellCommandExecutor.cancelApplication();
+    }
+
+    @Override
+    public AbstractParameters getParameters() {
+        return javaParameters;
+    }
+
+    /**
+     * @description: Replaces placeholders such as local variables in source files
+     * @param: [java.lang.String]
+     * @return: java.lang.String
+     **/
+    protected static String convertJavaSourceCodePlaceholders(String rawScript) throws StringIndexOutOfBoundsException {
+        int len = "${setShareVar(${".length();
+        int scriptStart = 0;
+        while ((scriptStart = rawScript.indexOf("${setShareVar(${", scriptStart)) != -1) {
+            int start = -1;
+            int end = rawScript.indexOf('}', scriptStart + len);
+            String prop = rawScript.substring(scriptStart + len, end);
+
+            start = rawScript.indexOf(',', end);
+            end = rawScript.indexOf(')', start);
+
+            String value = rawScript.substring(start + 1, end);
+
+            start = rawScript.indexOf('}', start) + 1;
+            end = rawScript.length();
+
+            String replaceScript = String.format("print(\"${{setValue({},{})}}\".format(\"%s\",%s))", prop, value);
+
+            rawScript = rawScript.substring(0, scriptStart) + replaceScript + rawScript.substring(start, end);
+
+            scriptStart += replaceScript.length();
+        }
+        return rawScript;
+    }
+
+    /**
+     * @description: Creates a Java source file when it does not exist
+     * @param: [java.lang.String, java.lang.String]
+     * @return: java.lang.String
+     **/
+    protected void createJavaSourceFileIfNotExists(String sourceCode, String fileName) throws IOException {
+        logger.info("tenantCode: {}, task dir:{}", taskRequest.getTenantCode(), taskRequest.getExecutePath());
+
+        if (!Files.exists(Paths.get(fileName))) {
+            logger.info("generate java source file: {}", fileName);
+
+            StringBuilder sb = new StringBuilder();
+            sb.append(sourceCode);
+            logger.info(sb.toString());
+
+            // write data to file
+            FileUtils.writeStringToFile(new File(fileName),
+                    sb.toString(),
+                    StandardCharsets.UTF_8);
```

Comment:
```suggestion
            logger.info("The java source code", sourceCode);

            // write data to file
            FileUtils.writeStringToFile(new File(fileName),
                    sb,
                    StandardCharsets.UTF_8);
```
Why do you use StringBuilder here?

Reply:
Thank you. The test code is not clean